src/TextInput/index.js

```diff
@@ -106,6 +109,8 @@ TextInput.propTypes = {
   startIcon: PropTypes.oneOf(VALID_ICON_NAMES),
   /** Name of Narmi icon to place at the end of the input box */
   endIcon: PropTypes.oneOf(VALID_ICON_NAMES),
+  /** Display an X at the end of label that clears input on click. Takes precedence over endIcon when input is not empty */
```

Comment:
> Takes precedence over endIcon when input is not empty

I was on my way here to say we should mention this. Thanks for writing good prop docs!